### PR TITLE
Add deployment options matrix

### PR DESCRIPTION
**Pull request by malomarrec**
We are missing a page for outlining what feature is supported on self-hosted, managed instances, or cloud only.
This is a proposal to add such a page.
[Slack discussion](https://sourcegraph.slack.com/archives/C01SD4A49E1/p1643966159528069)

### CODE REVIEW

**Comment by malomarrec** [comment]
TIL: need to add a schema.

**Comment by jasonyavorsky** [comment]
https://handbook.sourcegraph.com/handbook/editing/content-generation may be helpful.

File: schema/deployment_options.schema.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (jasonyavorsky) [inline schema/deployment_options.schema.json:1]: @malomarrec schema/features.schema.json should probably change too to account for the new possibilities in that file.

**Review by jasonyavorsky — APPROVED** [review]

File: schema/deployment_options.schema.json (hunk 1 above)
Comment (malomarrec) [reply schema/deployment_options.schema.json:1]: ty!

**Comment by malomarrec** [comment]
@Joelkw @macraig @ryphil - I am merging this, but feel free to edit the data files to represent Cloud availability. I don't think we need to be exhaustive about what _is_ supported where (that would create a huge, noisy page), rather we should highlight what is not supported.

**Comment by jasonyavorsky** [comment]
Really nice change @malomarrec thanks for this. Remember to share in some channel so people are aware.

**Pull request merged**